lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
+            warn "TIMEOUT parsing #{file}. Skipping."
+          rescue => e
```

Comment:
> Otherwise, I guess objections have been met and this should go in?

Sorry, I may have might be slightly unclear saying "fine with me to keep these in this PR...", or I'm misunderstanding what you're saying here 😃. I am not comfortable shipping a bare rescue that swallows all exceptions it catches as warnings.

What I meant was: if the change to rescue behavior is necessary to avoid an overall behavior change  (i.e. there is code that was parsed successfully on JRuby but not the new MRI) branch, then hashing out the right approach to the exception handling in this PR is fine. That applies to both this question around the RegExp errors and the timeout handling.